@xen-orchestra/audit-core/src/index.js

```diff
@@ -32,7 +32,8 @@ export const NULL_ID = 'nullId'
 
 const HASH_ALGORITHM_ID = '5'
 const createHash = (data, algorithmId = HASH_ALGORITHM_ID) =>
-  `$${algorithmId}$$${hash(data, {
+  // delete "undefined" properties and normalize data with JSON.stringify
+  `$${algorithmId}$$${hash(JSON.parse(JSON.stringify(data)), {
```

Comment:
Wouldn't it be better to do this outside and also pass the normalized data to the storage?